packages/legend-application-repl/src/stores/dataCube/grid/DataCubeGridState.ts

```diff
@@ -139,4 +148,166 @@ export class DataCubeGridState extends DataCubeQuerySnapshotSubscriber {
       await this.dataCube.replStore.client.getGridClientLicenseKey(),
     );
   }
+
+  generateCSVFile = () => {
+    const params = {
+      fileName: 'csv_data.csv',
+    };
+    if (this._client) {
+      this._client.exportDataAsCsv(params);
+    } else {
+      console.error('Grid API not set');
+    }
+  };
+
+  generateExcelFile = () => {
+    if (!this._client) {
+      throw new Error('Grid API is not initialized.');
+    }
+    const params = {
+      fileName: 'excel_data.xlsx',
+      processCellCallbacks: (cell: any) => {
+        return cell ? cell.toString() : '';
+      },
+    };
+    this._client.exportDataAsExcel(params);
+  };
+
+  blobToBase64 = (blob: Blob): Promise<string> => {
+    return new Promise((resolve, reject) => {
+      const reader = new FileReader();
+      reader.onloadend = () => {
+        const result = reader.result;
+        if (typeof result === 'string') {
+          const base64Data = result.split(',')[1];
+          if (base64Data) {
+            resolve(base64Data);
+          } else {
+            console.error('Base64 conversion failed: Invalid base64 data.');
+            reject(
+              new Error('Failed to convert blob to base64. Base64 is invalid.'),
+            );
+          }
+        } else {
+          console.error(
+            'Base64 conversion failed: FileReader result is not a string.',
+          );
+          reject(
+            new Error(
+              'Failed to convert blob to base64. FileReader result is not a string.',
+            ),
+          );
+        }
+      };
+      reader.onerror = (error) => {
+        console.error('Base64 conversion failed: FileReader error.', error);
+        reject(
+          new Error('Failed to convert blob to base64: FileReader error.'),
+        );
+      };
+      reader.readAsDataURL(blob);
+    });
+  };
+
+  generateEmail = async (isHtml: boolean, includeAttachments: boolean) => {
+    try {
+      const textContent =
+        'Please find the attached CSV and Excel(.xlsx) files.';
+      const htmlContent = `
+      <html>
+        <body>
+          <p>${textContent}</p>
+        <body>
+      </html>
+      `;
+
+      const emailBody = isHtml ? htmlContent : textContent;
+      const contentType = isHtml ? 'text/html' : 'text/plain';
+      const boundaryMixed = 'boundary----------123456789----------987654321';
+      const boundaryAlternative =
+        'boundary----------0987654321----------1234567890';
+
+      let emlContent = `
+From: somebody@test.com
```

Comment:
probably can leave the sender/receiver blank for now

Reply:
Done